File: app/src/main/java/me/devsaki/hentoid/enums/Theme.java
package me.devsaki.hentoid.enums;

import androidx.annotation.NonNull;

/**
 * App themes
 */
public enum Theme {

    // Warning : names are _resoure names_ that should match the xml files, not display names
    LIGHT(0, "Light"),
    DARK(1, "Dark"), // "Deep red" in settings
    BLACK(2, "Black"), // "Dark" in settings, not to be mistaken with usual "black" that means AMOLED
    NONE(99, "Light");

    private final int id;
    private final String resourceName;

    Theme(int id, @NonNull String resourceName) {
        this.id = id;
        this.resourceName = resourceName;
    }

    public int getId() {
        return id;
    }

    public String getResourceName() {
        return resourceName;
    }


    public static Theme searchById(int id) {
        for (Theme s : values())
            if (id == s.id) return s;

        return NONE;
    }
}

File: app/src/main/java/me/devsaki/hentoid/enums/Theme.kt
package me.devsaki.hentoid.enums

import me.devsaki.hentoid.R

enum class Theme(val id: Int, val resourceName: String, val nameRes : Int) {
    // Warning : names are _resoure names_ that should match the xml files, not display names
    LIGHT(0, "Light", R.string.theme_light),

    // "Deep red" in settings
    DARK(1, "Dark", R.string.theme_deep_red),

    // "Dark" in settings, not to be mistaken with usual "black" that means AMOLED
    BLACK(2, "Black", R.string.theme_dark),

    // Material You
    YOU(3, "You", R.string.theme_you),

    // Fallback
    NONE(99, "Light", R.string.none);

    companion object {
        fun searchById(id: Int): Theme {
            for (s in entries) if (id == s.id) return s
            return Theme.NONE
        }
    }
}